### PR TITLE
New package: PCSXR-PGXP

### PR DESCRIPTION
**Pull request by Vlad-Romanov**
This is enhance version of PCSXR, gives much better graphics

I know my templates not perfect please bare with me, I will fix all these tomorrow before I PR more. Thank you for be accommodating :D

### CODE REVIEW

File: srcpkgs/pcsxr-pgxp/template [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Template file for 'pcsxr-pgxp'
+pkgname=pcsxr-pgxp
+version=1.9.99
+revision=1
+lib32disabled=yes
+wrksrc="pcsxr-master"
+nostrip_files="pcsxr"
+build_style=cmake
+configure_args="-DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_INSTALL_LIBDIR=/usr/lib
```
Comment (cr6git) [inline srcpkgs/pcsxr-pgxp/template]: `-DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_INSTALL_LIBDIR=/usr/lib`

These are already set by default if you use `build_style=cmake`
See the script `common/build-style/cmake.sh`

File: srcpkgs/pcsxr-pgxp/template [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ -DSND_BACKEND=sdl -DENABLE_CCDDA=ON -DUSE_LIBARCHIVE=ON -DUSE_LIBCDIO=ON -DOpenGL_GL_PREFERENCE=GLVND"
+hostmakedepends="pkg-config automake libtool intltool glib-devel gettext-devel nasm zip unzip"
```
Comment (cr6git) [inline srcpkgs/pcsxr-pgxp/template]: I understand `unzip` to extract the archive, but are you sure `zip` is needed too?

File: srcpkgs/pcsxr-pgxp/template [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+makedepends="libXv-devel libXtst-devel libcdio-devel SDL-devel gtk+3-devel libarchive-devel
+ libXv-devel gdk-pixbuf-devel SDL2-devel SDL-devel libXext-devel libpulseaudio libX11-devel libXxf86vm
+ zlib-devel libepoxy-devel glu-devel MesaLib-devel ffmpeg ffmpeg-devel"
```
Comment (cr6git) [inline srcpkgs/pcsxr-pgxp/template]: I think it's `libpulseaudio-devel` , not `libpulseaudio`

The same for `ffmpeg`

File: srcpkgs/pcsxr-pgxp/template [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+depends="desktop-file-utils ffmpeg libarchive libcdio libXv gdk-pixbuf libGL libGLES SDL2 SDL libXext
+ libpulseaudio libX11 libXxf86vm zlib libepoxy glu libtxc_dxtn libavcodec libavdevice libavformat libavfilter
+ libavresample libswresample libswscale libavutil"
```
Comment (cr6git) [inline srcpkgs/pcsxr-pgxp/template]: These dependencies should be automatically detected...

File: srcpkgs/pcsxr-pgxp/template [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+short_desc="Sony PlayStation (PSX) emulator based on the PCSX-df project"
+maintainer="VladTheImpaler <vladimir-romanov-ii@yandex.com>"
+license="GPL-3"
```
Comment (cr6git) [inline srcpkgs/pcsxr-pgxp/template]: [SPDX](https://spdx.org/licenses/)

File: srcpkgs/pcsxr-pgxp/template [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+homepage="http://pcsxr.codeplex.com/"
+distfiles="https://github.com/icatbutler/pcsxr/archive/master.zip"
+checksum=a2a5a90b833db3827af5a4c1ed0b011b9a20474942e8357181ea3577b2ad0766
+nopie=yes
```
Comment (cr6git) [inline srcpkgs/pcsxr-pgxp/template]: Why `nopie=yes` ?

File: srcpkgs/pcsxr-pgxp/template [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+ libXv-devel gdk-pixbuf-devel SDL2-devel SDL-devel libXext-devel libpulseaudio libX11-devel libXxf86vm
+ zlib-devel libepoxy-devel glu-devel MesaLib-devel ffmpeg ffmpeg-devel"
+depends="desktop-file-utils ffmpeg libarchive libcdio libXv gdk-pixbuf libGL libGLES SDL2 SDL libXext
+ libpulseaudio libX11 libXxf86vm zlib libepoxy glu libtxc_dxtn libavcodec libavdevice libavformat libavfilter
+ libavresample libswresample libswscale libavutil"
+short_desc="Sony PlayStation (PSX) emulator based on the PCSX-df project"
+maintainer="VladTheImpaler <vladimir-romanov-ii@yandex.com>"
+license="GPL-3"
+homepage="http://pcsxr.codeplex.com/"
+distfiles="https://github.com/icatbutler/pcsxr/archive/master.zip"
```
Comment (cr6git) [inline srcpkgs/pcsxr-pgxp/template]: This is wrong! You should never fetch the master zip from github.
Use the release tarball, or a precise commit if tarball is not available.

File: srcpkgs/pcsxr-pgxp/template (hunk 2 above)
Comment (cr6git) [reply srcpkgs/pcsxr-pgxp/template]: See my comment below...

File: srcpkgs/pcsxr-pgxp/template (hunk 3 above)
Comment (Vlad-Romanov) [reply srcpkgs/pcsxr-pgxp/template]: there's pulseaudio-devel & libpulseaudio, no libpulseaudio-devel, at least not that I see...

File: srcpkgs/pcsxr-pgxp/template (hunk 4 above)
Comment (Vlad-Romanov) [reply srcpkgs/pcsxr-pgxp/template]: All of these? I figure they must be include during compile for OpenGL plugin, I will try to mess with dependencies tomorrow

File: srcpkgs/pcsxr-pgxp/template (hunk 6 above)
Comment (Vlad-Romanov) [reply srcpkgs/pcsxr-pgxp/template]: would not build without it, gave error.

File: srcpkgs/pcsxr-pgxp/template (hunk 7 above)
Comment (Vlad-Romanov) [reply srcpkgs/pcsxr-pgxp/template]: There is no release tarball... :/ is why I use master.zip

File: srcpkgs/pcsxr-pgxp/template (hunk 5 above)
Comment (Vlad-Romanov) [reply srcpkgs/pcsxr-pgxp/template]: I was using normal pcsxr template when made this, GPL-3 is what is in pcsxr template.

File: srcpkgs/pcsxr-pgxp/template (hunk 7 above)
Comment (cr6git) [reply srcpkgs/pcsxr-pgxp/template]: Are you sure about `version=1.9.99` ?
According to [ChangeLog](https://github.com/iCatButler/pcsxr/blob/master/ChangeLog) the latest seems to be 1.9.92.

Comment (cr6git) [reply srcpkgs/pcsxr-pgxp/template]: You can fetch the [latest commit](https://github.com/icatbutler/pcsxr/commit/7936d466c58f6f2603900fbbc9d18b56ac2e4b4c) by modifying your template:
````
version=1.9.92.20180427
_commit=7936d466c58f6f2603900fbbc9d18b56ac2e4b4c
wrksrc="pcsxr-${_commit}"

distfiles="https://github.com/icatbutler/pcsxr/archive/${_commit}.tar.gz"
checksum=5d7541abef8f8f42ca92bee4f7f6785bd8ea2dcf8eed65261cd13de798dfbe13
````
Should work.

**Comment by Vlad-Romanov** [comment]
hmmm.... @cr6git there is patch for pcsxr, how I can set patch to apply for only x86_64 & i686 but no ARM / MUSL ?

**Comment by cr6git** [comment]
@Vlad-Romanov To apply a patch only for these archs? You could use something like:
````
post_extract() {
	case $XBPS_TARGET_MACHINE in
		i686|x86_64) patch -Np0 -i ${FILESDIR}/your.patch ;;
		*) ;;
	esac
}
````

**Pull request closed without merging**